tasks/libs/common/gomodules.py

```diff
@@ -0,0 +1,323 @@
+
+        try:
+            output = subprocess.check_output(
+                ["go", "run", ".", "-path", os.path.join(base_path, self.path), "-prefix", AGENT_MODULE_PATH_PREFIX],
+                cwd=mod_parser_path,
+            ).decode("utf-8")
+        except subprocess.CalledProcessError as e:
+            print(f"Error while calling go.mod parser: {e.output}")
+            raise e
+
+        # Remove github.com/DataDog/datadog-agent/ from each line
+        return [line[len(AGENT_MODULE_PATH_PREFIX) :] for line in output.strip().splitlines()]
+
+    # FIXME: Change when Agent 6 and Agent 7 releases are decoupled
+    def tag(self, agent_version):
+        """Return the module tag name for a given Agent version.
+        >>> mods = [GoModule("."), GoModule("pkg/util/log")]
+        >>> [mod.tag("7.27.0") for mod in mods]
+        [["6.27.0", "7.27.0"], ["pkg/util/log/v0.27.0"]]
+        """
+        if self.path == ".":
+            return ["6" + agent_version[1:], "7" + agent_version[1:]]
+
+        return [f"{self.path}/{self.__version(agent_version)}"]
+
+    def full_path(self):
+        """Return the absolute path of the Go module."""
+        return str((self.base_dir / self.path).resolve())
+
+    def go_mod_path(self):
+        """Return the absolute path of the Go module go.mod file."""
+        return self.full_path() + "/go.mod"
+
+    @property
+    def dependencies(self):
+        if not self._dependencies:
+            self._dependencies = self.__compute_dependencies()
+        return self._dependencies
+
+    @property
+    def import_path(self):
+        """Return the Go import path of the Go module
+        >>> mods = [GoModule("."), GoModule("pkg/util/log")]
+        >>> [mod.import_path for mod in mods]
+        ["github.com/DataDog/datadog-agent", "github.com/DataDog/datadog-agent/pkg/util/log"]
+        """
+        path = AGENT_MODULE_PATH_PREFIX.removesuffix('/')
+        if self.path != ".":
+            path += "/" + self.path
+        return path
+
+    def dependency_path(self, agent_version):
+        """Return the versioned dependency path of the Go module
+        >>> mods = [GoModule("."), GoModule("pkg/util/log")]
+        >>> [mod.dependency_path("7.27.0") for mod in mods]
+        ["github.com/DataDog/datadog-agent@v7.27.0", "github.com/DataDog/datadog-agent/pkg/util/log@v0.27.0"]
+        """
+        return f"{self.import_path}@{self.__version(agent_version)}"
+
+
+AGENT_MODULE_PATH_PREFIX = "github.com/DataDog/datadog-agent/"
+
+
+def list_default_modules(base_dir: Path | None = None) -> tuple[dict[str, GoModule], set[str]]:
+    """List all the module configurations of the repository.
+
+    Returns:
+        A tuple of (modules, ignored_module_paths).
+    """
+
+    modules = {}
+    ignored_modules = set()
+
+    for gomod_path in glob("**/go.mod", recursive=True, root_dir=base_dir):
+        module_data_dir = Path(gomod_path).parent
+
+        module = GoModule.from_file(module_data_dir, base_dir)
```

Comment:
🥜 While this is clear and explicit, you could have used the module path as attribute for a "natural" constructor
```suggestion
        module = GoModule(gomod_path)
```

Reply:
This won't parse the file, `from_file` is used to parse the file and create the `GoModule` while `GoModule.__init__` is the dataclass constructor (mostly used to have default values / to create the object after parsing the file)